packages/clarity-js/src/data/upload.ts

```diff
@@ -187,8 +187,16 @@ function send(payload: string, zipped: Uint8Array, sequence: number, beacon: boo
             try {
                 // Navigator needs to be bound to sendBeacon before it is used to avoid errors in some browsers
                 dispatched = navigator.sendBeacon.bind(navigator)(url, payload);
-                if (dispatched) { done(sequence); }
-            } catch { /* do nothing - and we will automatically fallback to XHR below */ }
+                if (dispatched) { 
+                    done(sequence); 
+                } else {
+                    // If sendBeacon fails, we report the error and continue with XHR upload
+                    report(new Error(`${Constant.BeaconError}: ${payload.length}`));
```

Comment:
There are two cases: false return and error. Could we send the error information if that was chat happened?